packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { RouteComponentProps, useLocation } from '@reach/router';
+import { currentAccount } from '../../../lib/cache';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import { Integration, useAuthClient } from '../../../models';
+import AppLayout from '../../../components/AppLayout';
+import CardHeader from '../../../components/CardHeader';
+import ConfirmSignupCode from '.';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { LocationState } from './interfaces';
+
+const SignupConfirmCodeContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: LocationState;
```

Comment:
I think we may want to set the type for `keyFetchToken` and `unwrapBKey` to be possibly undefined in case users reach this screen without those in location state. We can probably put the checks for these the same place we check for `localStorage` and then pass them to the presentation layer if they're present.